### PR TITLE
Implement ftoa function in stdlib

### PR DESCRIPTION
**Pull request by niv256**
* Add files via upload

* double to string conversion

* Update ftoa.c

* Update objects.mk

* ftoa #2

* Delete ftoa.c

* Changed after comments

pow will be implemented in math.h

Co-authored-by: Niv Meiran <niv25600@gmail.com>

### CODE REVIEW

**Review by OfekMarks — APPROVED** [review]
Seems great

**Comment by idoSinai** [comment]
Did you add a function signature to stdio.h?

File: libc/stdlib/ftoa.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#include <math.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+int get_decimal_places(float number);
+
+char *ftoa(float number, int ndigit, char *buf) {
+  ndigit = fmin((uint32_t)ndigit, get_decimal_places(number));
```
Comment (idoSinai) [inline libc/stdlib/ftoa.c]: why did you convert ndigit to (uint32_t) but didn't convert get_decimal_places return value to (uint32_t) too?

Comment (niv256) [reply libc/stdlib/ftoa.c]: originally this was in order to discard negative ndigit values, but instead we it now takes it as a uint8_t type argument so it should be clearer.

File: libc/stdlib/ftoa.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    utoa(fraction, buf, 10);
+  }
+
+  return buf;
+}
+
+uint8_t get_decimal_places(float number) {
+  uint8_t count = 0;
+
+  while ((number = number * 10 - (int)number * 10)) {
```
Comment (idoSinai) [inline libc/stdlib/ftoa.c:35]: use uint8_t instead of int

**Pull request merged**